internal/eval/evalers.go

```diff
@@ -74,6 +74,18 @@ func evalLong(n Evaler, env *Env) (types.Long, error) {
 	return l, nil
 }
 
+func evalComparableValue(n Evaler, env *Env) (types.ComparableValue, error) {
+	v, err := n.Eval(env)
+	if err != nil {
+		return nil, err
+	}
+	l, ok := v.(types.ComparableValue)
```

Comment:
Can you make ComparableValue be an interface that only exists within the evalers package?  I don't think the general public needs to be aware of it.